jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/AbsoluteUnixPath.java

```diff
@@ -98,6 +98,42 @@ public AbsoluteUnixPath resolve(RelativeUnixPath relativeUnixPath) {
     return new AbsoluteUnixPath(newPathComponents.build());
   }
 
+  /**
+   * Resolves this path against another relative path (by the name elements of {@code
+   * relativePath}).
+   *
+   * @param relativePath the relative path to resolve against
+   * @return a new {@link AbsoluteUnixPath} representing the resolved path
+   */
+  // TODO: Add test.
+  public AbsoluteUnixPath resolve(Path relativePath) {
+    Preconditions.checkArgument(
+        relativePath.getRoot() == null, "Cannot resolve against absolute Path: " + relativePath);
+
+    ImmutableList.Builder<String> newPathComponents =
+        ImmutableList.builderWithExpectedSize(pathComponents.size() + relativePath.getNameCount());
+    newPathComponents.addAll(pathComponents);
+    for (Path pathComponent : relativePath) {
+      if (pathComponent.toString().isEmpty()) {
```

Comment:
You needed this when parsing `String` directly, but I don't think it's needed here. For example, `System.out.println(Paths.get("///a///a///a//").getNameCount())` return 3.

Reply:
I'm using this mostly for `Paths.get("")`, which would return a `nameCount` of 1, but also as a just-in-case since it's not guaranteed that name elements are not empty for different implementations of `Path`.

Reply:
Huh, TIL about `Paths.get("")` having an empty filename.

Reply:
However, since resolving against a `""` path doesn't add another name element, not having this in `fromPath` works.

Reply:
*having this in the call to `fromPath` in `resolve(Path relativePath)`